src/Page.re

```diff
@@ -46,3 +96,35 @@ type typeOptions = {. "delay": float};
 [@bs.send.pipe : t]
 external type_ : (string, string, ~options: typeOptions=?, unit) => Js.Promise.t(unit) =
   "type";
+
+[@bs.send.pipe : t] [@bs.splice]
+external deleteCookie : array(cookie) => Js.Promise.t(unit) = "";
+
+[@bs.send.pipe : t] [@bs.splice]
+external cookies : array(string) => Js.Promise.t(array(cookie)) = "";
+
+[@bs.send.pipe : t] [@bs.splice]
+external setCookie : array(cookie) => Js.Promise.t(unit) = "";
+
+[@bs.send.pipe : t]
+external emulate : emulateOption => Js.Promise.t(unit) = "";
+
+[@bs.send.pipe : t] external viewport : unit => viewport = "";
+
+[@bs.send]
+external _emulateMedia : (t, Js.Nullable.t(string)) => Js.Promise.t(unit) =
+  "emulateMedia";
+
+type mediaType =
+  | Screen
+  | Print;
+
+let emulateMedia = (mediaType: option(mediaType), t) =>
```

Comment:
This could all be boiled down to
```reason
[@bs.send.pipe : t]
external emulateMedia :
  (~mediaType: Js.nullable([@bs.string] [ | `screen | `print])=?) =>
  Js.Promise.t(unit) =
  "";
```
Then call with ``page |> Page.emulateMedia(`screen)``.

Reply:
oops, the code will fail:
```
  370 ┆ testPromise("emulateMedia()", () =>
  371 ┆   Js.Promise.(
  372 ┆     page^ |> Page.emulateMedia(`print) |> then_(() => pass |> resolve
        )
  373 ┆   )
  374 ┆ );

  This has type:
    [> `print ]
  But somewhere wanted:
    BsPuppeteer.Page.t (defined as BsPuppeteer.FrameBase.t)
```

Reply:
Oops yeah, we want to be able to call it like that. I guess it should not be optional.

Reply:
I'm thinking about separate binding: 
```ml
[@bs.send.pipe : t]
external emulateMedia :
  ([@bs.string] [ | `screen | `print]) => Js.Promise.t(unit) =
  "";

[@bs.send.pipe : t]
external emulateMediaNull : ([@bs.as {json|null|json}] _) => Js.Promise.t(unit) =
  "";
```
And add test case for each:
```ml
  testPromise("emulateMedia()", () =>
    Js.Promise.(
      page^ |> Page.emulateMedia(`print) |> then_(() => pass |> resolve)
    )
  );
  testPromise("emulateMediaNull()", () =>
    Js.Promise.(page^ |> Page.emulateMediaNull |> then_(() => pass |> resolve))
  );
```

(We need better naming for `emulateMediaNull`)

Reply:
Yeah I was thinking similarly. What if we call the null one emulateMediaDisable, which is what it is doing. You need to set the `= "emulateMedia"` at the end of that external.

Reply:
Fixed in commit 4d6b40b